tensorflow/java/src/gen/cc/java_defs.h

```diff
@@ -0,0 +1,266 @@
+enum Modifier {
+  PUBLIC    = (1 << 0),
+  PROTECTED = (1 << 1),
+  PRIVATE   = (1 << 2),
+  STATIC    = (1 << 3),
+  FINAL     = (1 << 4),
+};
+
+class Annotation;
+
+/// \brief A definition of any kind of Java type (classes, interfaces...)
+///
+/// Note that most of the data fields of this class are only useful in specific
+/// contexts and are not required in many cases. For example, annotations and
+/// supertypes are only useful when declaring a type.
+class Type {
+ public:
+  enum Kind {
+    PRIMITIVE, CLASS, INTERFACE, ENUM, GENERIC, ANNOTATION
+  };
+  struct Comparator {
+    bool operator() (const Type& type1, const Type& type2) {
+      return type1.name_ < type2.name_ || type1.package_ < type2.package_;
+    }
+  };
+  static Type Primitive(const string& name) {
+    return Type(Type::PRIMITIVE, name, "");
+  }
+  static Type Class(const string& name, const string& package = "") {
+    return Type(Type::CLASS, name, package);
+  }
+  static Type Interface(const string& name, const string& package = "") {
+    return Type(Type::INTERFACE, name, package);
+  }
+  static Type Enum(const string& name, const string& package = "") {
+    return Type(Type::ENUM, name, package);
+  }
+  static Type Generic(const string& name) {
+    return Type(Type::GENERIC, name, "");
+  }
+  static Type Wildcard() {
+    return Type(Type::GENERIC, "", "");
+  }
+  static Type ClassOf(const Type& type) {
+    return Class("Class").param(type);
+  }
+  static Type ListOf(const Type& type) {
+    return Interface("List", "java.util").param(type);
+  }
+  static Type IterableOf(const Type& type) {
+    return Interface("Iterable").param(type);
+  }
+  const Kind& kind() const { return kind_; }
+  const string& name() const { return name_; }
+  const string& package() const { return package_; }
+  const string& descr() const { return descr_; }
+  Type& descr(const string& descr) { descr_ = descr; return *this; }
+  const std::vector<Type>& params() const { return params_; }
+  Type& param(const Type& param) {
+    params_.push_back(param);
+    return *this;
+  }
+  const std::vector<Annotation>& annotations() const { return annotations_; }
+  Type& annotation(const Annotation& annotation) {
+    annotations_.push_back(annotation);
+    return *this;
+  }
+  const std::deque<Type>& supertypes() const { return supertypes_; }
+  Type& supertype(const Type& type) {
+    if (type.kind_ == CLASS) {
+      supertypes_.push_front(type);  // keep superclass at the front of the list
+    } else if (type.kind_ == INTERFACE) {
+      supertypes_.push_back(type);
+    }
+    return *this;
+  }
+  /// Returns true if "type" is of a known collection type (only a few for now)
+  bool IsCollection() const {
+    return name_ == "List" || name_ == "Iterable";
+  }
+  /// Returns true if this instance is a wildcard (<?>)
+  bool IsWildcard() const {
+    return kind_ == GENERIC && name_.empty();
+  }
+  bool operator==(const Type& type) const {
```

Comment:
Why do we need the operator overloads?

Reply:
Like the comparator, the usage of the == operator will come in a next PR so I'll remove it for now